src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -271,6 +293,8 @@ public void writeTo(DownloadInstance di, Class<?> clazz, Type type, Annotation[]
 
                             if ("original".equals(di.getConversionParamValue())) {
                                 logger.fine("stored original of an ingested file requested");
+                                // Get orginal file size before storageIO is reassigned to prevent an NPE.
+                                fileSize = storageIO.getDataFile().getOriginalFileSize();
```

Comment:
Note that switching the `fileSize` to the original size is not enough here, if the size of the tabular file has already been used to calculate the offset for a "last N bytes" ("bytes=-N") range type.

Reply:
Thankfully!